### PR TITLE
wallet-connectors: Remove deprecated JSON-RPC client integration

### PR DESCRIPTION
**Pull request by bisgardo**
## Purpose

Remove support for the depreated JSON-RPC client as per [this strategy](https://concordium.atlassian.net/wiki/spaces/EN/pages/1185349645/Remove+V1+GRPC+API).

## Changes

Remove all symbols related to `WalletConnection.getJsonRpcClient`.

### CODE REVIEW

File: packages/wallet-connectors/CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 -   `WalletConnect`: Fix schema object format conversion in `signAndSendTransaction` in request payloads.
 -   `WalletConnect`: Use standard string identifiers for transaction type in request payload.
 
+### Removed
+
```
Comment (DOBEN) [inline packages/wallet-connectors/CHANGELOG.md:20]: Please update the `READMe.md` file in `react-components` package. The file mentions the `jsonRpcUrl`.

File: packages/wallet-connectors/CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+-   `WalletConnection`: The deprecated method `getJsonRpcClient()` and related symbols has been removed
+    in favor of the new gRPC client (`ConcordiumGRPCClient`) for the Node API v2.
+    Existing usage is migrated by simply constructing and using this client instead of the one fetched with `connection.getJsonRpcClient`.
+    The `Network` type has a field `grpcOpts` which may be used to construct the client (see its docstring for details).
+    The field is optional, but present on the predefined constants `TESTNET` and `MAINNET`.
+    For users of `@concordium/react-components`, this all wrapped into the hook `useGrpcClient`.
+
```
Comment (DOBEN) [inline packages/wallet-connectors/CHANGELOG.md:27]: There is the `JsonRpcClient` remaining in `useContractSelector` file which should be replaced:

```
./react-components/src/useContractSelector.ts
```

The comments only mention the `JSON-RPC` but not the GRPC option. 
` * @param rpc JSON-RPC proxy client through which to perform the lookup.`

File: packages/wallet-connectors/CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 -   `WalletConnect`: Fix schema object format conversion in `signAndSendTransaction` in request payloads.
 -   `WalletConnect`: Use standard string identifiers for transaction type in request payload.
 
+### Removed
+
+-   `WalletConnection`: The deprecated method `getJsonRpcClient()` and related symbols has been removed
```
Comment (DOBEN) [inline packages/wallet-connectors/CHANGELOG.md:21]:
```suggestion
-   `WalletConnection`: The deprecated method `getJsonRpcClient()` and related symbols have been removed
```

File: packages/wallet-connectors/CHANGELOG.md [hunk 4]
```diff
@@ -16,6 +16,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+    in favor of the new gRPC client (`ConcordiumGRPCClient`) for the Node API v2.
+    Existing usage is migrated by simply constructing and using this client instead of the one fetched with `connection.getJsonRpcClient`.
+    The `Network` type has a field `grpcOpts` which may be used to construct the client (see its docstring for details).
+    The field is optional, but present on the predefined constants `TESTNET` and `MAINNET`.
+    For users of `@concordium/react-components`, this all wrapped into the hook `useGrpcClient`.
```
Comment (DOBEN) [inline packages/wallet-connectors/CHANGELOG.md:26]:
```suggestion
    For users of `@concordium/react-components`, this is all wrapped into the hook `useGrpcClient`.
```

File: packages/wallet-connectors/CHANGELOG.md (hunk 2 above)
Comment (bisgardo) [reply packages/wallet-connectors/CHANGELOG.md:27]: Nice catch, will definitely update the docs. The option of passing `JsonRpcClient` doesn't add any complexity, so I'm inclined to think that should keep the param type as an option for as long as the type is present in the web SDK. Are there plans for removing the type from there soon?

Comment (bisgardo) [reply packages/wallet-connectors/CHANGELOG.md:27]: updated docstring

File: packages/wallet-connectors/CHANGELOG.md [hunk 5]
```diff
@@ -16,6 +16,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 -   `WalletConnect`: Fix schema object format conversion in `signAndSendTransaction` in request payloads.
 -   `WalletConnect`: Use standard string identifiers for transaction type in request payload.
 
+### Removed
+
+-   `WalletConnection`: The deprecated method `getJsonRpcClient()` and related symbols have been removed
+    in favor of the new gRPC client (`ConcordiumGRPCClient`) for the Node API v2.
+    Existing usage is migrated by simply constructing and using this client instead of the one fetched with `connection.getJsonRpcClient`.
```
Comment (bisgardo) [inline packages/wallet-connectors/CHANGELOG.md:23]: Could add that
```typescript
withJsonRpcClient(connection, func)
```
can be replaced by just
```typescript
func(rpc)
```

Comment (bisgardo) [reply packages/wallet-connectors/CHANGELOG.md]: added

File: packages/react-components/src/useContractSelector.ts [hunk 6]
```diff
@@ -101,7 +101,7 @@ export interface ContractSelector {
 
 /**
  * React hook to look up a smart contract's data and state from its index.
- * @param rpc JSON-RPC proxy client through which to perform the lookup.
+ * @param rpc gRPC client through which to perform the lookup. The JSON-RPC client is supported for backwards compatibility only.
```
Comment (bisgardo) [inline packages/react-components/src/useContractSelector.ts:104]: Support for `JsonRpcClient` will be removed once we upgrade to a version of `web-sdk` that doesn't include the type.

**Comment by bisgardo** [comment]
ping @DOBEN @orhoj @shjortConcordium

**Review by orhoj — APPROVED** [review]

**Comment by orhoj** [comment]
LGTM

**Review by soerenbf — APPROVED** [review]

**Pull request merged**